File: Examples/VBasic/GroupDocs.Search.Examples.VBasic/CustomFieldExtractor.vb

'ExStart:Customfieldextractor
Class CustomFieldExtractor
    Implements IFieldExtractor
    Public ReadOnly Property Extensions() As String() Implements IFieldExtractor.Extensions
        'redefine internal field extractor
        Get
            Return {".doc", ".doc"}
        End Get

        'extractor for supporting new document format
        'Get
        '    Return {".doc", ".doc"}
        'End Get
    End Property
    Public Function GetFields(fileName As String) As FieldInfo() Implements IFieldExtractor.GetFields
        Dim result As FieldInfo() = New FieldInfo(3) {}
        result(0) = New FieldInfo("Content", "Hardcoded document content")
        result(1) = New FieldInfo("DocumentType", "MyDocumentType")
        result(2) = New FieldInfo("Author", "Hardcoded author")
        result(3) = New FieldInfo("CreationDate", "21.05.2004")
        Return result
    End Function
End Class
'ExEnd:Customfieldextractor

File: Examples/CustomFieldExtractor.cs
using GroupDocs.Search;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Search_for_.NET
{
    //ExStart:Customfieldextractor
    class CustomFieldExtractor : IFieldExtractor
    {
        public string[] Extensions
        {
            //redefine internal field extractor
            get { return new[] {".doc", ".doc"}; }
            //extractor for supporting new document format
            //get { return new[] { ".ext1", ".ext2" }; }
        }

        public FieldInfo[] GetFields(string fileName)
        {
            FieldInfo[] result = new FieldInfo[4];
            result[0] = new FieldInfo("Content", "Hardcoded document content");
            result[1] = new FieldInfo("DocumentType", "MyDocumentType");
            result[2] = new FieldInfo("Author", "Hardcoded author");
            result[3] = new FieldInfo("CreationDate", "21.05.2004");
            return result;
        }
    }
    //ExEnd:Customfieldextractor
}
